src/NumericalAlgorithms/DiscontinuousGalerkin/NormalDotFlux.hpp

```diff
@@ -74,6 +74,53 @@ auto normal_dot_flux(
 }
 /// @}
 
+/// @{
+/*!
+ * \brief Multiplies a surface normal covector with a flux tensor or variables
+ *
+ * \details
+ * Returns the outer product $n_j v_{k\ldots}$, where $n_j$ is the `normal` and
+ * $v_{k\ldots}$ is the `flux_tensor`.
```

Comment:
This sounds like it would be useful as an eager-math outer product, rather than hiding it in DiscontinuousGalerkin.

Reply:
Added an eager-math outer product. I kept the `normal_times_flux` function because of the docs and the Variables overload and the similarity to `normal_dot_flux`.

Reply:
The Variables overload is a reasonable argument.  The tensor `normal_dot_flux` should also be in eager math, so I wouldn't use it as inspiration, but that's not necessary for this PR.